applications/MPMApplication/custom_conditions/particle_based_conditions/mpm_particle_base_dirichlet_condition.cpp

```diff
@@ -66,6 +66,74 @@ void MPMParticleBaseDirichletCondition::FinalizeSolutionStep( const ProcessInfo&
     m_imposed_displacement = ZeroVector(3);
 }
 
+void MPMParticleBaseDirichletCondition::CalculateInterfaceContactForce(const ProcessInfo& rCurrentProcessInfo )
+{
+    const unsigned int number_of_nodes = GetGeometry().PointsNumber();
+
+    GeneralVariables Variables;
+    MPMShapeFunctionPointValues(Variables.N);
+
+    array_1d<double, 3 > mpc_force = ZeroVector(3);
+
+    for ( unsigned int i = 0; i < number_of_nodes; i++ )
+    {
+        if (Variables.N[i] > std::numeric_limits<double>::epsilon() )
+        {
+            auto r_geometry = GetGeometry();
+            
+            const double nodal_area  = r_geometry[i].FastGetSolutionStepValue(NODAL_AREA, 0);
+            const Vector nodal_force = r_geometry[i].FastGetSolutionStepValue(REACTION);
+
+            if (nodal_area > std::numeric_limits<double>::epsilon())
+            {
+                mpc_force += Variables.N[i] * nodal_force * this->GetIntegrationWeight() / nodal_area;
+            }
+        }
+    }
+
+    // Apply in the normal contact direction and allow releasing motion
+    if (Is(CONTACT))
+    {
+        // Apply only in the normal direction
+        const double normal_force = MathUtils<double>::Dot(mpc_force, m_normal);
+
+        // This check is done to avoid sticking forces
+        if (normal_force > 0.0)
+            mpc_force = 1.0 * normal_force * m_normal;
+        else
+            mpc_force = ZeroVector(3);
+    }
+    
+    m_contact_force = -mpc_force;
+}
+
+void MPMParticleBaseDirichletCondition::CalculateNodalReactions(const ProcessInfo& rCurrentProcessInfo )
+{
+    KRATOS_ERROR << "You are calling the CalculateNodalReactions from the base class for dirichlet conditions" << std::endl;
+}
+
+
+void MPMParticleBaseDirichletCondition::CalculateOnIntegrationPoints(
+    const Variable<bool>& rVariable,
+    std::vector<bool>& rValues,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    if (rValues.size() != 1)
+        rValues.resize(1);
```

Comment:
[`std::vector::resize`](https://en.cppreference.com/w/cpp/container/vector/resize) does this check internally.

```suggestion
    rValues.resize(1);
```